nntrainer/layers/time_dist.cpp

```diff
@@ -60,7 +122,68 @@ int TimeDistLayer::initialize(Manager &manager) {
 }
 
 void TimeDistLayer::forwarding(bool training) {
-  // NYI
+  setPosition();
+
+  Tensor &hidden_ = net_hidden[0]->getVariableRef();
+  Tensor &input_ = net_input[0]->getVariableRef();
+  // input_.dim = [ b, 1, h, w ]
+
+  Tensor hidden_g, h_g;
+
+  TensorDim ho_dim = hidden_.getDim();
+  TensorDim in_dim = input_.getDim();
+
+  // TODO: This transposed Input Tensor could be resued for backwarding
+  Tensor in = transposeTensor(input_);
+
+  Tensor out = Tensor({ho_dim[2], 1, ho_dim[0], ho_dim[3]}, true);
+
+  TensorDim i_dim = in_dim;
+  i_dim.channel(1);
+  i_dim.height(1);
+
+  TensorDim h_dim = ho_dim;
+  h_dim.channel(1);
+  h_dim.height(1);
+
+  if (dist_layer->getType() == "loss") {
+    hidden_g = net_hidden[0]->getGradientRef();
+    if (!hidden_g.uninitialized()) {
+      h_g = transposeTensor(hidden_g);
+    }
+  }
+
+  for (unsigned int i = 0; i < in_dim.height(); ++i) {
+    //
+    // Iterate Height Direction. The dimension of in is input_[ b, 1, 1, width
+    // ]. The dimension of out is hidden_[ b, 1, 1, width ]
+    //
+    Tensor label_iter;
+
+    Tensor in_iter =
+      in.getSharedDataTensor(i_dim, i * in_dim.batch() * in_dim.width());
+    Tensor out_iter =
+      out.getSharedDataTensor(h_dim, i * ho_dim.batch() * ho_dim.width());
+
+    Var_Grad in_var(i_dim, true, false, dist_layer->getName() + ":input");
```

Comment:
this is minor.
how about create these outside the loop, and just initialize them again and again?

Reply:
right. Thanks